### PR TITLE
packages: add nvidia OCI hook for peer pods

### PR DESCRIPTION
**Pull request by msanft**
This enables the `podvm-image` package for use in GPU peer pods by doing the following:
- Adding a preconfigured `nvidia-ctk` package for use with peer pods GPU containers.
- Adding a package for the OCI hook required to start peer pods GPU containers.
- Placing the OCI hook in the expected path in the podvm-image if GPU use is configured.

### CODE REVIEW

File: packages/by-name/mkNixosConfig/package.nix [hunk 1]
```diff
@@ -35,15 +35,24 @@ lib.makeOverridable (
...
+          {
+            inherit (outerPkgs)
+              azure-no-agent
+              cloud-api-adaptor
+              kernel-podvm-azure
+              pause-bundle
+              ;
+            inherit (outerPkgs.kata) kata-agent;
+          }
+          // lib.optionalAttrs config.contrast.gpu.enable {
```
Comment (Freax13) [inline packages/by-name/mkNixosConfig/package.nix:49]: Do we need to make the packages conditional on the `gpu.enable`? Isn't everything here lazily evaluated anyway?

**Comment by katexochen** [comment]
```
       error: evaluation aborted with the following error message: 'lib.customisation.callPackageWith: Function called without required argument "libnvidia-container-custom" at /nix/store/dvppxran2w1d1jgs7lhrih6012vmd24l-source/packages/by-name/nvidia-ctk-with-config/package.nix:9'
```

I guess this depends on another PR? Wrong base?

File: packages/by-name/mkNixosConfig/package.nix (hunk 1 above)
Comment (Freax13) [reply packages/by-name/mkNixosConfig/package.nix:49]: @msanft Did we resolve this offline? I can't remember.

Comment (msanft) [reply packages/by-name/mkNixosConfig/package.nix]: Sorry, forgot to push :P

**Review by Freax13 — APPROVED** [review]

**Review by katexochen — APPROVED** [review]

**Pull request merged**